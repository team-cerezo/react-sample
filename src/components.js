import React from 'react';

export const InputContent = (props) => {
    const content = props.content;
    const updateContent = props.updateContent;
    const tryAddTodo = props.tryAddTodo;
    return (
        <p>
            <input type="text" placeholder="やることある？" value={content}
                onChange={updateContent} onKeyPress={tryAddTodo} />
        </p>
    );
};

export const TodoComponent = (props) => {
    const todo = props.todo;
    const updateStatus = props.updateStatus;
    return (
        <li>
            <label>
                <input type="checkbox" checked={todo.done}
                    onChange={event => updateStatus(todo.id, event.target.checked)} />
                <span>#{todo.id} {todo.content}</span>
            </label>
        </li>
    );
};

export const TodoListComponent = (props) => (
    <ul>
        {props.children}
    </ul>
);

export const ClearButton = (props) => (
    <p>
        <button onClick={props.clear}>終わったやつ消す</button>
    </p>
);

