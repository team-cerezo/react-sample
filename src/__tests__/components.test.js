import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { InputContent, TodoComponent, TodoListComponent, ClearButton } from '../components';

Enzyme.configure({ adapter: new Adapter() });

describe('InputContent', () => {

    test('描画', () => {
        const component = renderer.create(
            <InputContent
                content={'Hello, world!'}
                updateContent={() => { }}
                tryAddTodo={() => { }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('内容が変更されたらupdateContentハンドラが呼ばれる', () => {
        const handler1 = jest.fn();
        const handler2 = jest.fn();
        const input = shallow(
            <InputContent
                content={''}
                updateContent={handler1}
                tryAddTodo={handler2} />
        );
        input.find('input').simulate('change', { target: { value: 'test' } });
        expect(handler1).toBeCalled();
        expect(handler2).not.toBeCalled();
    });

    test('Enterキーが押されたらtryAddTodoハンドラが呼ばれる', () => {
        const handler1 = jest.fn();
        const handler2 = jest.fn();
        const input = shallow(
            <InputContent
                content={''}
                updateContent={handler1}
                tryAddTodo={handler2} />
        );
        input.find('input').simulate('keypress', { key: 'Enter' });
        expect(handler1).not.toBeCalled();
        expect(handler2).toBeCalled();
    });

    test('Enterキー以外のキーが押されたらtryAddTodoハンドラが呼ばれる', () => {
        const handler1 = jest.fn();
        const handler2 = jest.fn();
        const input = shallow(
            <InputContent
                content={''}
                updateContent={handler1}
                tryAddTodo={handler2} />
        );
        input.find('input').simulate('keypress', { key: 'NotEnter' });
        expect(handler1).not.toBeCalled();
        expect(handler2).not.toBeCalled();
    });
});

describe('TodoComponent', () => {

    test('描画', () => {
        const todo = { id: 123, conetnt: 'foobar', done: true };
        const updateStatus = (id, done) => { };
        const component = renderer.create(
            <TodoComponent
                todo={todo}
                updateStatus={updateStatus} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('チェックのON/OFFが変更されたらupdateStatusハンドラが呼ばれる', () => {
        const todo = { id: 123, conetnt: 'foobar', done: true };
        const handler = jest.fn();
        const input = shallow(
            <TodoComponent
                todo={todo}
                updateStatus={handler} />
        );
        const event = { target: { checked: false } };
        input.find('input').simulate('change', event);
        expect(handler).toBeCalledWith(todo.id, event.target.checked);
    });
});

describe('TodoListComponent', () => {

    test('描画', () => {
        const component = renderer.create(
            <TodoListComponent>
                {['foo', 'bar', 'baz'].map(x => <li key={x}>{x}</li>)}
            </TodoListComponent>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ClearButton', () => {

    test('描画', () => {
        const component = renderer.create(
            <ClearButton clear={() => { }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ボタンを押したらclearハンドラが呼ばれる', () => {
        const handler = jest.fn();
        const button = shallow(
            <ClearButton clear={handler} />
        );
        button.find('button').simulate('click');
        expect(handler).toBeCalled();
    });
});
