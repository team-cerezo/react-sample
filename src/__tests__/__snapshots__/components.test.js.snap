// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton 描画 1`] = `
<p>
  <button
    onClick={[Function]}
  >
    終わったやつ消す
  </button>
</p>
`;

exports[`InputContent 描画 1`] = `
<p>
  <input
    autoFocus={true}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="やることある？"
    type="text"
    value="Hello, world!"
  />
</p>
`;

exports[`TodoComponent 描画 1`] = `
<li>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span>
      #
      123
       
    </span>
  </label>
</li>
`;

exports[`TodoListComponent 描画 1`] = `
<ul>
  <li>
    foo
  </li>
  <li>
    bar
  </li>
  <li>
    baz
  </li>
</ul>
`;
